rasa/core/events/__init__.py

```diff
@@ -660,59 +679,92 @@ def as_dict(self) -> Dict[Text, Any]:
     def _from_story_string(cls, parameters: Dict[Text, Any]) -> Optional[List[Event]]:
 
         trigger_date_time = parser.parse(parameters.get("date_time"))
+
         return [
             ReminderScheduled(
-                parameters.get("action"),
+                parameters.get("intent"),
                 trigger_date_time,
-                parameters.get("name", None),
-                parameters.get("kill_on_user_msg", True),
-                parameters.get("timestamp"),
-                parameters.get("metadata"),
+                parameters.get("entities"),
+                name=parameters.get("name"),
+                kill_on_user_message=parameters.get("kill_on_user_msg", True),
+                timestamp=parameters.get("timestamp"),
+                metadata=parameters.get("metadata"),
             )
         ]
 
 
 # noinspection PyProtectedMember
 class ReminderCancelled(Event):
-    """Cancel all jobs with a specific name."""
+    """Cancel certain jobs."""
 
     type_name = "cancel_reminder"
 
     def __init__(
         self,
-        action_name: Text,
+        name: Optional[Text] = None,
+        intent: Optional[Text] = None,
+        entities: Optional[List[Dict]] = None,
         timestamp: Optional[float] = None,
         metadata: Optional[Dict[Text, Any]] = None,
     ):
-        """
+        """Creates a ReminderCancelled event.
+
+        All arguments are optional and used to refine the set of reminders that
+        are to be cancelled. If no arguments are supplied, this will cancel all reminders.
+
         Args:
-            action_name: name of the scheduled action to be cancelled
-            metadata: optional event metadata
+            name: Name of the reminder to be cancelled.
+            intent: Intent name that is to be used to identify the reminders to be cancelled.
+            entities: Entities that are to be used to identify the reminders to be cancelled.
+            timestamp: Optional timestamp.
+            metadata: Optional event metadata.
         """
 
-        self.action_name = action_name
+        self.name = name
+        self.intent = intent
+        self.entities = entities
         super().__init__(timestamp, metadata)
 
     def __hash__(self) -> int:
-        return hash(self.action_name)
+        return hash(
+            (
+                self.name,
+                self.intent,
+                self.entities,
+            )
+        )
 
     def __eq__(self, other) -> bool:
         return isinstance(other, ReminderCancelled)
 
     def __str__(self) -> Text:
-        return f"ReminderCancelled(action: {self.action_name})"
+        return f"ReminderCancelled(name: {self.name}, intent: {self.intent}, entities: {self.entities})"
+
+    def matches_job_string(self, job_string: Text, sender_id: Text) -> bool:
```

Comment:
can you please add a comment what's happening here? I'd also suggest to rename the method to reflect the intent of it (see suggestion in other comment)